espnet2/enh/layers/dpmulcat.py

```diff
@@ -0,0 +1,168 @@
+import torch
+import torch.nn as nn
+
+
+class MulCatBlock(nn.Module):
+    """The MulCat block that multiplies two RNN layers' outputs
+        and concatenate it with the input Tensor.
+
+    Args:
+        input_size: int, dimension of the input feature. The input should have shape
+            (batch, seq_len, input_size).
+        hidden_size: int, dimension of the hidden state.
+        dropout: float, the dropout rate in the LSTM layer. (Default: 0.0)
+        bidirectional: bool, whether the RNN layers are bidirectional. (Default: True)
+    """
+
+    def __init__(
+        self,
+        input_size: int,
+        hidden_size: int,
+        dropout: float = 0.0,
+        bidirectional: bool = True,
+    ):
+        super(MulCatBlock, self).__init__()
+
+        num_direction = int(bidirectional) + 1
+
+        self.rnn = nn.LSTM(
+            input_size,
+            hidden_size,
+            1,
+            dropout=dropout,
+            batch_first=True,
+            bidirectional=bidirectional,
+        )
+        self.rnn_proj = nn.Linear(hidden_size * num_direction, input_size)
+
+        self.gate_rnn = nn.LSTM(
+            input_size,
+            hidden_size,
+            num_layers=1,
+            batch_first=True,
+            dropout=dropout,
+            bidirectional=bidirectional,
+        )
+        self.gate_rnn_proj = nn.Linear(hidden_size * num_direction, input_size)
+
+        self.block_projection = nn.Linear(input_size * 2, input_size)
+
+    def forward(self, input):
```

Comment:
Could you add a docstring to describe the expected input and output shapes? It would greatly help in understanding how to use this module.